src/components/ImageUpload.js

```diff
@@ -35,32 +35,43 @@ export class ImageUpload extends Component {
         console.log(error);
       },
       () => {
-        // complete function ....
         storage
           .ref("images")
           .child(image.name)
           .getDownloadURL()
           .then(url => {
+            const plantsRef = firebase.database().ref("plants");
+            const plant = {
+              images: {
+                url: this.state.url
```

Comment:
I think you should use the `url` from line `42` here not the `this.state.url` since you set state below.